### PR TITLE
[Task #52212] [Auth] Register

### PR DESCRIPTION
**Pull request by thonght-3051**
## Related Tickets
- [#52212](https://edu-redmine.sun-asterisk.vn/issues/52212)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

### CODE REVIEW

**Comment by dangthibichngocb** [comment]
Nhiều commit với nhiều file change thế anh

**Review by dangthibichngocb — APPROVED** [review]

**Comment by thonght-3051** [comment]
ready

File: app/controllers/auth_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class AuthController < ApplicationController
+  layout "auth"
+  def register
```
Comment (ghost) [inline app/controllers/auth_controller.rb]: để là new đi em

File: app/controllers/auth_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    @user = User.new
+    @address = @user.addresses.new
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      redirect_to root_url
+    else
+      render :register
```
Comment (ghost) [inline app/controllers/auth_controller.rb]: Thêm flash vào

File: db/migrate/20220815043146_add_index_to_email.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToEmail < ActiveRecord::Migration[6.1]
+  def change
+    add_index :users, :email, unique: true
+  end
```
Comment (ghost) [inline db/migrate/20220815043146_add_index_to_email.rb]: gộp lại thành 1 file đi em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by thonght-3051** [comment]
ready

**Pull request merged**